### DISCUSSION TITLE
Help with regular expression in contains

### DISCUSSION THREAD

**Discussion by igorgiumellizup** [cypress-io/cypress#17292] [Questions and Help] [answered]
Hello

I have a page with two buttons:

PopToViewInvalidRoute
PopToView

Both button tittles have spaces in the beginning and at the end. I'm using the following strategy to try to click the second button, PopToView, but it is returning the first.

cy.contains('button', new RegExp('\s?' + 'PopToView' + '\s?')).click()

Any ideas on how to click the second button? Thanks

**Comment by bahmutov** [comment, marked as answer]
You would need to add `^` and `$` around the text and optional white space to make sure there is nothing else, see examples in https://glebbahmutov.com/cypress-examples/7.7.0/commands/querying.html#cy-contains-with-regular-expression
